test/e2e/util.go

```diff
@@ -1264,6 +1303,38 @@ func routeToNode(nodeName string, ips []string, mtu int, add bool) error {
 	return nil
 }
 
+// GetNodeIPv6LinkLocalAddressForEth0 returns the IPv6 link-local address for eth0 interface
+func GetNodeIPv6LinkLocalAddressForEth0(nodeName string) (string, error) {
+	// Command to get IPv6 link-local address for eth0
+	ipCmd := []string{"ip", "-6", "addr", "show", "dev", "eth0", "scope", "link"}
+
+	cmd := []string{"docker", "exec", nodeName}
+	cmd = append(cmd, ipCmd...)
+
+	output, err := runCommand(cmd...)
+	if err != nil {
+		return "", fmt.Errorf("failed to get link-local address for eth0: %v", err)
+	}
+
+	// Parse the output to extract the fe80:: address
+	lines := strings.Split(string(output), "\n")
+
+	for _, line := range lines {
+		if strings.Contains(line, "inet6 fe80::") {
```

Comment:
cant you extract the ip and use netlink lib to detect if LL?

Reply:
yup yup good catch

Reply:
done.